lib/index.js

```diff
@@ -289,8 +302,7 @@ internals.getParamsData = function (param, name, typeName) {
         meta: param.meta,
         unit: param.unit,
         type: type,
-        allowedValues: type !== 'reference' && param.valids ? internals.getExistsValues(type, param.valids) : null,
-        disallowedValues: type !== 'reference' && param.invalids ? internals.getExistsValues(type, param.invalids) : null,
+        disallowedValues: param.invalids ? internals.getExistsValues(originalType, param.invalids) : null,
```

Comment:
I honestly can't remember why I made this change. I suspect that a coverage complaint was the mostly likely case but am not certain.